pontoon/administration/files.py

```diff
@@ -64,27 +77,56 @@ def split_comments_mine(
     if (not end or (end > pointer)) and len(text) > pointer:
         cls.split_entities(text, object, code=code, pointer=pointer)
 
-PropertiesParser.split_comments = split_comments_mine
+PropertiesParser.split_comments = split_comments_pro
+
+
+@classmethod
+def split_comments_ini(cls, text, object, pointer=0, end=None):
+    pattern = cls.patterns['comment']
+    if end:
+        match = pattern.search(text, pointer, end)
+    else:
+        match = pattern.search(text, pointer)
+    while match:
+        st0 = match.start(0)
+        if st0 > pointer:
+            cls.split_entities(text, object, pointer=pointer, end=st0)
+        comment = Comment()
+        comment.add(
+            match.group(0)[1:].replace('\n;', '\n').replace('\n#', '\n'))
+        object.append(comment)
+        pointer = match.end(0)
+        if end:
+            match = pattern.search(text, pointer, end)
+        else:
+            match = pattern.search(text, pointer)
+    if (not end or (end > pointer)) and len(text) > pointer:
+        cls.split_entities(text, object, pointer=pointer, end=end)
+
+IniParser.split_comments = split_comments_ini
+
+IniParser.patterns['entity'] = re.compile(
+    '^[ \t]*([^#!\s\n][^=:\n]*?)[ \t]*[:=][ \t]*(.*?)(?<!\\\)(?=\n|\Z)',
+    re.S | re.M)
 
 
 def __repr__mine(self):
-        string = ''
```

Comment:
I see in https://github.com/mathjazz/pontoon/commit/a816e7d48383fbc33c1cb7fa7901bd48006ba2f3 that we're using this in `extract_silme`. If that's the only spot, I'd recommend avoiding monkey-patching the class and just using this code directly in that method. If we're the only consumer there's no reason to patch.

Also, a generator and `join` would be a slightly faster, more pythonic way of doing this:

```python
return ''.join(unicode(i) for i in self)
```